Character.cpp

```diff
@@ -86,32 +87,45 @@ int Character::takeDamage(int damage)
     return hitPoints;
 }
 
-
-#include <assert>
 void Character::attackInternal(Character& other)
 {
-    if( other.hitPoints <= 0 )
+    if(other.hitPoints <= 0)
     {
         /*
         When you defeat another Character: 
             a) your stats are restored to their initial value
             b) your stats are boosted 10%
             c) the initial value of your stats is updated to reflect this boosted stat for the next time you defeat another character.
       */
-        assert(false);
+        //assert(false);
+        // a)
+        if(hitPoints < *initialHitPoints) { hitPoints = *initialHitPoints; }
+        if(armor < *initialArmorLevel) { armor = *initialArmorLevel; }
+        if(attackDamage < *initialAttackDamage) { attackDamage = *initialAttackDamage; }
+        // b)
+        boostHitPoints(hitPoints * 0.1);
+        boostArmor(armor * 0.1);
+        boostAttackDamage(attackDamage * 0.1); 
```

Comment:
Good, but you can just do `val *= 1.1;` instead of calling these functions.
Honestly, who knows how the compiler would optimize that.